src/lib.rs

```diff
@@ -381,4 +381,51 @@ impl Contract {
 
         lockup_ids
     }
+
+    pub fn discard_draft_group(&mut self, draft_group_id: DraftGroupIndex) {
+        self.assert_deposit_whitelist(&env::predecessor_account_id());
+
+        let mut draft_group = self
+            .draft_groups
+            .get(&draft_group_id as _)
+            .expect("draft group not found");
+        draft_group.discard();
+
+        if draft_group.draft_indices.is_empty() {
+            self.draft_groups.remove(&draft_group_id as _);
+        } else {
+            self.draft_groups.insert(&draft_group_id as _, &draft_group);
+        }
+    }
+
+    pub fn delete_drafts(&mut self, draft_ids: Vec<DraftIndex>) {
+        let mut draft_group_lookup: HashMap<DraftGroupIndex, DraftGroup> = HashMap::new();
+        draft_ids.iter().for_each(|draft_id| {
+            let draft = self.drafts.remove(&draft_id as _).expect("draft not found");
+            let draft_group = draft_group_lookup
+                .entry(draft.draft_group_id)
+                .or_insert_with(|| {
+                    self.draft_groups
+                        .get(&draft.draft_group_id as _)
+                        .expect("draft group not found")
+                });
+
+            draft_group.assert_can_delete_draft();
+            let amount = draft.total_balance();
+            assert!(draft_group.total_amount >= amount, "Invariant");
+            draft_group.total_amount -= amount;
+
+            assert!(draft_group.draft_indices.remove(draft_id), "Invariant");
+        });
+
+        draft_group_lookup
+            .iter()
+            .for_each(|(draft_group_id, draft_group)| {
+                if draft_group.draft_indices.is_empty() {
+                    self.draft_groups.remove(&draft_group_id as _);
+                } else {
+                    self.draft_groups.insert(&draft_group_id as _, &draft_group);
+                }
+            });
```

Comment:
`for_each` is rarely justified as it most of the cases `for` loop will be much easier to read:

```suggestion
        for (draft_group_id, draft_group) in &draft_group_lookup {
            if draft_group.draft_indices.is_empty() {
                self.draft_groups.remove(&draft_group_id as _);
            } else {
                self.draft_groups.insert(&draft_group_id as _, &draft_group);
            }
        }
```